app/src/main/java/com/github/swent/echo/compose/authentication/createProfile.kt

```diff
@@ -0,0 +1,251 @@
+                Spacer(modifier = Modifier.width(5.dp))
+
+                // Add tag button
+                SmallFloatingActionButton(
+                    onClick = {},
+                    containerColor = MaterialTheme.colorScheme.secondaryContainer,
+                    contentColor = MaterialTheme.colorScheme.secondary,
+                    modifier = Modifier.testTag("AddTag")
+                ) {
+                    Icon(Icons.Default.Add, "Add tags")
+                }
+            }
+
+            Spacer(modifier = Modifier.weight(1f))
+
+            // Save button
+            OutlinedButton(
+                onClick = { /* viewModel.profilesave() */},
+                modifier = Modifier.fillMaxWidth().testTag("Save")
+            ) {
+                Text(text = stringResource(id = R.string.profile_creation_save_button))
+            }
+        }
+    }
+}
+
+@Composable
+fun DropDownListFunctionWrapper(elementList: List<Any>, label: Int) {
+    var showDropdown by rememberSaveable { mutableStateOf(false) }
+    var selectedField by remember { mutableStateOf("") }
+    var selectedFieldSize by remember { mutableStateOf(Size.Zero) }
+    val icon = if (showDropdown) Icons.Filled.KeyboardArrowUp else Icons.Filled.KeyboardArrowDown
+    Column {
+        Box() {
+            OutlinedTextField(
+                value = selectedField,
+                onValueChange = { selectedField = it },
+                modifier =
+                    Modifier.onGloballyPositioned { coordinates ->
+                            selectedFieldSize = coordinates.size.toSize()
+                        }
+                        .clickable { showDropdown = !showDropdown }
+                        .testTag(stringResource(id = label)),
+                readOnly = true,
+                label = { Text(stringResource(id = label)) },
+                trailingIcon = {
+                    Icon(icon, "list dropdown", Modifier.clickable { showDropdown = !showDropdown })
+                }
+            )
+            DropdownMenu(
+                properties = PopupProperties(focusable = false),
+                expanded = showDropdown,
+                onDismissRequest = { showDropdown = false },
+                modifier =
+                    Modifier.align(Alignment.TopStart)
+                        .heightIn(max = 200.dp)
+                        .widthIn(with(LocalDensity.current) { selectedFieldSize.width.toDp() }),
+                offset = DpOffset(0.dp, 0.dp)
+            ) {
+                elementList.forEach { elem ->
+                    DropdownMenuItem(
+                        text = { Text(elem.toString()) },
+                        onClick = {
+                            selectedField = elem.toString()
+                            showDropdown = false
+                        },
+                        modifier = Modifier.testTag(elem.toString())
+                    )
+                }
+            }
+        }
+    }
+}
+
```

Comment:
Don't forget to comment what each Composable are doing, this one wasn't really clear at a first sight